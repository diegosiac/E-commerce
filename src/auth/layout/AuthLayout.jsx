import { Grid, Typography } from '@mui/material'
import PropTypes from 'prop-types'

export const AuthLayout = ({ children, title }) => {
  return (
    <Grid
      item
      component='main'
      bgcolor='grey.50'
      className='max-w-[90%] p-6 sm:w-[450px] rounded-lg my-12'
    >
      <Typography variant='h2' component='h1' className='mb-2'>{title}</Typography>
      {children}
    </Grid>
  )
}

AuthLayout.defaultProps = {
  title: ''
}

AuthLayout.propTypes = {
  children: PropTypes.node,
  title: PropTypes.string
}
