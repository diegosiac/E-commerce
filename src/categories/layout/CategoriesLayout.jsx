import { Box, Grid, Typography } from '@mui/material'
import Proptypes from 'prop-types'
import { CardContend } from '../components/CardContend'
import hardware from '../../assets/imgs/backgrounds/hardware.webp'

export const CategoriesLayout = ({
  children,
  title,
  disableGrid,
  urlSecondary,
  titlePrimary,
  titleSecondary,
  titleThird,
  titleFourth,
  descriptionPrimary,
  descriptionSecondary,
  descriptionThird,
  descriptionFourth
}) => {
  return (
    <Grid
      container
      className='justify-center flex-col items-center mb-16'
    >

      <Grid
        container
        className='w-11/12 mt-5 mb-14'
      >
        <Box className='flex mb-8'>
          <Box className='w-2 h-full bg-[#3483fa] mr-2' />
          <Typography variant='h3' component='h2'>{title}</Typography>
        </Box>

        <Grid
          container
          className='justify-evenly gap-6'
        >
          {children}
        </Grid>

      </Grid>

      {
        !disableGrid &&
          <Box
            className='grid grid-cols-2 gap-4 w-11/12 md:grid-cols-4'
          >

            <CardContend
              classImg='object-cover h-[200px]'
              urlImg={hardware}
              title={titlePrimary}
              description={descriptionPrimary}
              className='col-span-2 md:row-span-2'
            />

            <CardContend
              classImg='object-contain w-32'
              urlImg={urlSecondary}
              title={titleSecondary}
              description={descriptionSecondary}
              className='flex col-span-2'
            />

            <CardContend
              title={titleThird}
              description={descriptionThird}
            />

            <CardContend
              title={titleFourth}
              description={descriptionFourth}
            />

          </Box>
      }

    </Grid>
  )
}

CategoriesLayout.proptypes = {
  children: Proptypes.node.isRequired,
  title: Proptypes.string.isRequired,
  disableGrid: Proptypes.bool,
  urlSecondary: Proptypes.string,
  titlePrimary: Proptypes.string,
  titleSecondary: Proptypes.string,
  titleThird: Proptypes.string,
  titleFourth: Proptypes.string,
  descriptionPrimary: Proptypes.string,
  descriptionSecondary: Proptypes.string,
  descriptionThird: Proptypes.string,
  descriptionFourth: Proptypes.string
}

CategoriesLayout.defaultProps = {
  disableGrid: false
}
