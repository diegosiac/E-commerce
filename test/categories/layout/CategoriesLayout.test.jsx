import { render } from '@testing-library/react'
import { CategoriesLayout } from '../../../src/categories/layout/CategoriesLayout'
import phoneConfig from '../../assets/imgs/backgrounds/phoneConfig.webp'

describe('Testing on <CategoriesLayout/>', () => {
  test('It must render correctly with the grid and match with the snapshot', () => {
    const tree = render(
      <CategoriesLayout
        title='Title test'
        titlePrimary='Title primary test'
        titleSecondary='Title secondary test'
        titleThird='Title third test'
        titleFourth='Title fourth test'
        descriptionPrimary='Lorem ipsum dolor sit amet consectetur adipisicing elit. Explicabo eligendi praesentium'
        descriptionSecondary='Lorem ipsum dolor sit amet consectetur adipisicing elit. Explicabo eligendi praesentium'
        descriptionThird='Lorem ipsum dolor sit amet consectetur adipisicing elit. Explicabo eligendi praesentium'
        descriptionFourth='Lorem ipsum dolor sit amet consectetur adipisicing elit. Explicabo eligendi praesentium'
        urlSecondary={phoneConfig}
      >
        test children
      </CategoriesLayout>
    )

    expect(tree).toMatchSnapshot()
  })

  test('should render correctly without including the grid and match with the snapshot', () => {
    const tree = render(
      <CategoriesLayout
        title='Title test'
        disableGrid
      >
        test children
      </CategoriesLayout>
    )

    expect(tree).toMatchSnapshot()
  })
})
