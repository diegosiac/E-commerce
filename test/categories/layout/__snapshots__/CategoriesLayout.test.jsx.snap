// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on <CategoriesLayout/> It must render correctly with the grid and match with the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container justify-center flex-col items-center mb-16 css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container w-11/12 mt-5 mb-14 css-11lq3yg-MuiGrid-root"
        >
          <div
            class="flex mb-8 MuiBox-root css-0"
          >
            <div
              class="w-2 h-full bg-[#3483fa] mr-2 MuiBox-root css-0"
            />
            <h2
              class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
            >
              Title test
            </h2>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container justify-evenly gap-6 css-11lq3yg-MuiGrid-root"
          >
            test children
          </div>
        </div>
        <div
          class="grid grid-cols-2 gap-4 w-11/12 md:grid-cols-4 MuiBox-root css-0"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root col-span-2 md:row-span-2 css-bhp9pd-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root flex flex-col justify-center css-46bh2p-MuiCardContent-root"
            >
              <h5
                class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
              >
                Title primary test
              </h5>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Explicabo eligendi praesentium
              </p>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root flex col-span-2 css-bhp9pd-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root flex flex-col justify-center css-46bh2p-MuiCardContent-root"
            >
              <h5
                class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
              >
                Title secondary test
              </h5>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Explicabo eligendi praesentium
              </p>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root flex flex-col justify-center css-46bh2p-MuiCardContent-root"
            >
              <h5
                class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
              >
                Title third test
              </h5>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Explicabo eligendi praesentium
              </p>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root flex flex-col justify-center css-46bh2p-MuiCardContent-root"
            >
              <h5
                class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
              >
                Title fourth test
              </h5>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Explicabo eligendi praesentium
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container justify-center flex-col items-center mb-16 css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container w-11/12 mt-5 mb-14 css-11lq3yg-MuiGrid-root"
      >
        <div
          class="flex mb-8 MuiBox-root css-0"
        >
          <div
            class="w-2 h-full bg-[#3483fa] mr-2 MuiBox-root css-0"
          />
          <h2
            class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
          >
            Title test
          </h2>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container justify-evenly gap-6 css-11lq3yg-MuiGrid-root"
        >
          test children
        </div>
      </div>
      <div
        class="grid grid-cols-2 gap-4 w-11/12 md:grid-cols-4 MuiBox-root css-0"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root col-span-2 md:row-span-2 css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root flex flex-col justify-center css-46bh2p-MuiCardContent-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
            >
              Title primary test
            </h5>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Explicabo eligendi praesentium
            </p>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root flex col-span-2 css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root flex flex-col justify-center css-46bh2p-MuiCardContent-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
            >
              Title secondary test
            </h5>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Explicabo eligendi praesentium
            </p>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root flex flex-col justify-center css-46bh2p-MuiCardContent-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
            >
              Title third test
            </h5>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Explicabo eligendi praesentium
            </p>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root flex flex-col justify-center css-46bh2p-MuiCardContent-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
            >
              Title fourth test
            </h5>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Explicabo eligendi praesentium
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing on <CategoriesLayout/> should render correctly without including the grid and match with the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container justify-center flex-col items-center mb-16 css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container w-11/12 mt-5 mb-14 css-11lq3yg-MuiGrid-root"
        >
          <div
            class="flex mb-8 MuiBox-root css-0"
          >
            <div
              class="w-2 h-full bg-[#3483fa] mr-2 MuiBox-root css-0"
            />
            <h2
              class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
            >
              Title test
            </h2>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container justify-evenly gap-6 css-11lq3yg-MuiGrid-root"
          >
            test children
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container justify-center flex-col items-center mb-16 css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container w-11/12 mt-5 mb-14 css-11lq3yg-MuiGrid-root"
      >
        <div
          class="flex mb-8 MuiBox-root css-0"
        >
          <div
            class="w-2 h-full bg-[#3483fa] mr-2 MuiBox-root css-0"
          />
          <h2
            class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
          >
            Title test
          </h2>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container justify-evenly gap-6 css-11lq3yg-MuiGrid-root"
        >
          test children
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
